ui/src/experiment/ExperimentsRouter.js

```diff
@@ -0,0 +1,55 @@
+import React from "react";
+import {
+  EuiPage,
+  EuiPageBody,
+  EuiPageContent,
+  EuiPageHeader,
+  EuiPageHeaderSection,
+  EuiText,
+} from "@elastic/eui";
+import { PageTitle } from "../components/page/PageTitle";
+
+import useDynamicScript from "../hooks/useDynamicScript";
+import loadComponent from "../utils/remoteComponent";
+import { defaultExperimentEngine } from "../config";
+
+export const ExperimentsRouter = ({ projectId }) => {
+  // Retrieve script from host dynamically
+  const { ready, failed } = useDynamicScript({
+    url: defaultExperimentEngine.url,
+  });
+
+  if (!ready || failed) {
+    return (
+      <EuiPage>
+        <EuiPageBody>
+          <EuiPageHeader>
+            <EuiPageHeaderSection>
+              <PageTitle title="Experiments" />
+            </EuiPageHeaderSection>
+          </EuiPageHeader>
+          <EuiPageContent>
+            <EuiText size="s" color="subdued">
+              {!defaultExperimentEngine.url
+                ? "No default Experiment Engine configured"
+                : !ready
+                ? "Loading Experiment Exgine ..."
+                : "Failed to load Experiment Exgine"}
+            </EuiText>
+          </EuiPageContent>
+        </EuiPageBody>
+      </EuiPage>
+    );
+  }
+
+  // Load component from remote host
+  const Component = React.lazy(
+    loadComponent(defaultExperimentEngine.appName, "./ExperimentsLandingPage")
+  );
+
+  return (
+    <React.Suspense fallback="Loading Experiments">
```

Comment:
What's the difference between this `fallback="Loading Experiments"` and `"Loading Experiment Exgine ..."` at L:36? In what scenarios each of them will be shown?

Reply:
`Loading Experiment Engine` happens when the `remoteEntry.js` file is being retrieved and appended to the html head.  This part is now required because we are not setting the remote module's configurations in `craco.config.js` at build time anymore.

`fallback="Loading Experiments"` or later for the router form,  may be `fallback="Loading Experiment Configuration"` is a placeholder while React lazy-loads the component (using `loadComponent` in the preceding lines). It's hardly noticeable but still required - without the `fallback` property, the rendering will fail in the first attempt.

Refactored the file to use `<FallbackView/>` for all scenarios where the remote component is not successfully loaded.